File: Selenium/Activities/Activity10.py
from selenium import webdriver
from selenium.webdriver.common.by import By

with webdriver.Firefox() as driver:
    driver.get("https://training-support.net/webelements/mouse-events")
    print(f"The page title is : {driver.title}")
    confirmTxtElement = driver.find_element(By.XPATH,"//p[@id = 'result']")
    lockBtn= driver.find_element(By.XPATH,"//h1[text() = 'Cargo.lock']/ancestor::div[@role = 'button']")
    tomlBtn = driver.find_element(By.XPATH,"//h1[text() = 'Cargo.toml']/ancestor::div[@role = 'button']")
    srcBtn = driver.find_element(By.XPATH,"//h1[text() = 'src']/ancestor::div[@role = 'button']")
    targetBtn = driver.find_element(By.XPATH,"//h1[text() = 'target']/ancestor::div[@role = 'button']")
			
    builder = webdriver.ActionChains(driver)
    builder.click(lockBtn).pause(5).move_to_element(tomlBtn).click(tomlBtn).perform()
    print(f"The confirmation text is : {confirmTxtElement.text}")
    builder.double_click(srcBtn).pause(2).move_to_element(targetBtn).context_click(targetBtn).perform()
    print(f"The confirmation text is : {confirmTxtElement.text}")
    driver.quit()

File: Selenium/Activities/Activity10.java
package examples;

import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

public class Activity10 {

	public static void main(String[] args) {

			WebDriver driver = new FirefoxDriver();
			
			driver.get("https://training-support.net/webelements/mouse-events");
			System.out.println("The page title is : "+ driver.getTitle());
			
			WebElement confirmTxtElement = driver.findElement(By.xpath("//p[@id = 'result']"));
			WebElement lockBtn= driver.findElement(By.xpath("//h1[text() = 'Cargo.lock']/ancestor::div[@role = 'button']"));
			WebElement tomlBtn = driver.findElement(By.xpath("//h1[text() = 'Cargo.toml']/ancestor::div[@role = 'button']"));
			WebElement srcBtn = driver.findElement(By.xpath("//h1[text() = 'src']/ancestor::div[@role = 'button']"));
			WebElement targetBtn = driver.findElement(By.xpath("//h1[text() = 'target']/ancestor::div[@role = 'button']"));
			
			Actions builder = new Actions(driver);
			
			builder.click(lockBtn).pause(5000).moveToElement(tomlBtn).click(tomlBtn).build().perform();
			System.out.println("The confirmation text is : "+confirmTxtElement.getText());
			
			builder.doubleClick(srcBtn).pause(2000).moveToElement(targetBtn).contextClick(targetBtn).build().perform();
			System.out.println("The confirmation text is : "+confirmTxtElement.getText());
			driver.quit();

	}

}
